framework/src/main/java/org/checkerframework/framework/type/treeannotator/LiteralTreeAnnotator.java

```diff
@@ -13,39 +13,29 @@
 import java.util.Set;
 import java.util.regex.Pattern;
 import javax.lang.model.element.AnnotationMirror;
-import org.checkerframework.framework.qual.ImplicitFor;
 import org.checkerframework.framework.qual.LiteralKind;
+import org.checkerframework.framework.qual.QualifierForLiterals;
 import org.checkerframework.framework.type.AnnotatedTypeFactory;
 import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.GenericAnnotatedTypeFactory;
 import org.checkerframework.framework.type.QualifierHierarchy;
 import org.checkerframework.framework.type.typeannotator.ImplicitsTypeAnnotator;
 import org.checkerframework.javacutil.AnnotationBuilder;
 import org.checkerframework.javacutil.BugInCF;
 
 /**
- * Adds annotations to a type based on the contents of a tree. By default, this class honors the
- * {@link org.checkerframework.framework.qual.ImplicitFor} annotation and applies implicit
- * annotations specified by {@link org.checkerframework.framework.qual.ImplicitFor} for any tree
- * whose visitor is not overridden or does not call {@code super}; it is designed to be invoked from
- * {@link
- * org.checkerframework.framework.type.AnnotatedTypeFactory#addComputedTypeAnnotations(javax.lang.model.element.Element,
- * org.checkerframework.framework.type.AnnotatedTypeMirror)} and {@link
- * org.checkerframework.framework.type.AnnotatedTypeFactory#addComputedTypeAnnotations(com.sun.source.tree.Tree,
- * org.checkerframework.framework.type.AnnotatedTypeMirror)}.
+ * Adds annotations to a type based on the contents of a tree. This class applies annotations
+ * specified by {@link org.checkerframework.framework.qual.QualifierForLiterals}; it is designed to
+ * be add to a {@link ListTreeAnnotator} via {@link
+ * GenericAnnotatedTypeFactory#createTreeAnnotator()}
  *
- * <p>{@link ImplicitsTreeAnnotator} does not traverse trees deeply by default.
+ * <p>{@link LiteralTreeAnnotator} does not traverse trees deeply.
  *
- * <p>This class takes care of three of the attributes of {@link
- * org.checkerframework.framework.qual.ImplicitFor}; the others are handled in {@link
- * org.checkerframework.framework.type.typeannotator.ImplicitsTypeAnnotator}.
- *
- * <p>TODO: we currently don't check that any attribute is set, that is, a qualifier could be
- * annotated as @ImplicitFor(), which might be misleading.
- *
- * @see org.checkerframework.framework.type.typeannotator.ImplicitsTypeAnnotator
  * @see TreeAnnotator
  */
-public class ImplicitsTreeAnnotator extends TreeAnnotator {
+//  TODO: we currently don't check that any attribute is set, that is, a qualifier could be
+// annotated as @QualifierForLiterals(), which might be misleading.
```

Comment:
Should `@QualifierForLiterals()` simply mean `@QualfierForLiterals(LiteralKind.ALL)`?